### PR TITLE
Drop dependencies and other updates

### LINKED ISSUE #31: Find a way to drop metaviz from dependencies
It causes lots of issues with installation on GSC and should be dropped from dependency list

### LINKED ISSUE #34: make utils check-compliant
Ensure the package passes `devtools::check()`

### LINKED ISSUE #35: Setup github actions for this package
self-explanatory

### LINKED ISSUE #37: drop maftools dependency
the maftools is used as dependency in 2 functions:
- `genome_to_exome` as a supported input format for incoming maf. As we are moving away from maftools we will discontinue support of this input format
- `sanitize_maf_data` as a mean to create onco matrix. This could be swapped for the function in helpers child

This will significantly reduce the dependency burden on installing this package.

### LINKED ISSUE #38: drop Complexheatmap dependency
The Complexheatmap is only used to generate an optional output in cnvKompare. The heatmap is not clustered. This can be substituted with ggplot2 implementation to decrease the number of dependencies.

### LINKED ISSUE #39: drop plyr dependency
There is only one place where `plyr` is used, and it is to round to the nearest 5. We can avoid requiring this dependency by using base round and specifying base of 5 like so:
```
base*round(x/base)

# examples
> 5*round(94.3/5)
[1] 95
> 5*round(92.3/5)
[1] 90
```
This will help to decrease the dependency load

### PR DESCRIPTION
**Pull request by Kdreval**
This is to address the #31 
Example of the new plot is in [this](https://morinlabsfu.slack.com/archives/C0224H120CU/p1710373860259219) Slack thred

### CODE REVIEW

**Comment by vladimirsouza** [comment]
In `FtestCNV` examples, could we give an example for the `metadata` parameter?

**Comment by Kdreval** [comment]
@vladimirsouza Done! :heavy_check_mark:

File: R/sanitize_maf_data.R [hunk 1]
```diff
@@ -47,9 +48,18 @@ sanitize_maf_data = function(mutation_maf_path,
...
-
-  maftools::oncoplot(maf_o, genes = genes_keep, writeMatrix = T, removeNonMutated = F)  #writes to working directory
+  maf_o = as.data.frame(mutation_maf_data)
+  onco_matrix <- create_onco_matrix(
+    maf_df = maf_o,
+    genes = genes_keep
+  ) %>% as.data.frame
+  #writes to working directory
+  write.table(
+    om,
```
Comment (vladimirsouza) [inline R/sanitize_maf_data.R]:
```
> safe_oncomatrix_path = sanitize_maf_data(
+  mutation_maf_data = GAMBLR.data::sample_data$grch37$maf,
+  genes_keep = c("MYC", "ID3", "ARID1A", "FOXO1", "TP53", "FAT4", "IGLL5")
+ )
There are no missing genes during onco matrix assembly.
Error in write.table(om, file = paste0(getwd(), "/onco_matrix.txt"), quote = F,  : 
  object 'om' not found
```

Comment (vladimirsouza) [reply R/sanitize_maf_data.R]: Also, could you indent the examples?

Comment (Kdreval) [reply R/sanitize_maf_data.R]: These both are fixed in the latest commit :+1:

Comment (vladimirsouza) [reply R/sanitize_maf_data.R]: After running the function, I got this:
```
There are no missing genes during onco matrix assembly.
your data is in: /home/vsouza/repos/gamblr_repos/gambl_results_new_test_exp_data/onco_matrix.tsv
```
However, the the extension of the created file is txt. Could you change `tsv` to `txt` in the output message and in the returned string?

Comment (Kdreval) [reply R/sanitize_maf_data.R]: Done! :heavy_check_mark:

**Review by vladimirsouza — APPROVED** [review]

**Pull request merged**